cluster/rpc/server.go

```diff
@@ -156,7 +156,11 @@ func (s *Server) Open() error {
 				grpc_sentry.UnaryServerInterceptor(),
 			)))
 	}
-	options = append(options, monitoring.InstrumentGrpc(*s.metrics)...)
+
+	if s.metrics != nil {
```

Comment:
this additional check is there to check if monitoring is enabled or not. Ideally I want to do it via `prometheus.DefaultRegistry= no-op` and keep initializing the metrics as is without any posibility of `nil`. But need some refactoring. Looking at my other comment